File: web/src/api/channel.ts
import { alovaInstance } from '../lib/alova'

export interface Channel {
  id: string
  name: string
  type: number
  type_name: string
  base_url: string
  status: number
  priority: number
  weight: number
  balance: number
  models: string[]
  model_mapping: Record<string, string> | null
  groups: string[]
  used_quota: number
  request_count: number
  success_count: number
  success_rate: number
  config: Record<string, any> | null
  created_at: string
}

export interface ChannelListResponse {
  total: number
  page: number
  size: number
  items: Channel[]
}

export interface CreateChannelRequest {
  name: string
  type: number
  base_url: string
  api_key: string
  models: string[]
  model_mapping?: Record<string, string>
  priority?: number
  weight?: number
  groups?: string[]
  config?: Record<string, any>
}

export interface ChannelTestResult {
  success: boolean
  latency_ms: number
  tested_at: string
}

export const channelApi = {
  list: (params?: { page?: number; size?: number; status?: number; type?: number }) =>
    alovaInstance.Get<ChannelListResponse>('/admin/channels', { params }),
  
  create: (data: CreateChannelRequest) =>
    alovaInstance.Post<Channel>('/admin/channels', data),
  
  update: (id: string, data: Partial<CreateChannelRequest>) =>
    alovaInstance.Put<Channel>(`/admin/channels/${id}`, data),
  
  delete: (id: string) =>
    alovaInstance.Delete(`/admin/channels/${id}`),
  
  test: (id: string) =>
    alovaInstance.Post<ChannelTestResult>(`/admin/channels/${id}/test`),

  updateStatus: (id: string, enabled: boolean) =>
    alovaInstance.Put<Channel>(`/admin/channels/${id}/status`, { enabled }),
  
  batchCreate: (channels: CreateChannelRequest[]) =>
    alovaInstance.Post<{ created: number; channels: Channel[] }>('/admin/channels/batch', { channels }),
}

File: internal/channel/domain/channel.go
package domain

import (
	"time"
)

type Channel struct {
	ID           string    `json:"id" xorm:"'id' uuid pk notnull"`
	Name         string    `json:"name" xorm:"varchar(128) notnull"`
	Type         int       `json:"type" xorm:"smallint notnull"`
	BaseURL      string    `json:"base_url" xorm:"'base_url' varchar(512) notnull"`
	APIKeyEnc    string    `json:"-" xorm:"'api_key_enc' text notnull"`
	Status       int       `json:"status" xorm:"smallint default(1)"`
	Priority     int       `json:"priority" xorm:"int default(0)"`
	Weight       int       `json:"weight" xorm:"int default(1)"`
	Balance      float64   `json:"balance" xorm:"decimal(12,4)"`
	Models       string    `json:"models" xorm:"text notnull"`
	ModelMapping string    `json:"model_mapping" xorm:"text"`
	Groups       string    `json:"groups" xorm:"text default('["default"]')"`
	UsedQuota    int64     `json:"used_quota" xorm:"bigint default(0)"`
	RequestCount int64     `json:"request_count" xorm:"bigint default(0)"`
	SuccessCount int64     `json:"success_count" xorm:"bigint default(0)"`
	Config       string    `json:"config" xorm:"text"`
	CreatedAt    time.Time `json:"created_at" xorm:"created"`
	UpdatedAt    time.Time `json:"updated_at" xorm:"updated"`
}

func (Channel) TableName() string {
	return "channels"
}

type ChannelRepository interface {
	Create(channel *Channel) error
	GetByID(id string) (*Channel, error)
	Update(channel *Channel) error
	Delete(id string) error
	List(page, size int, status, channelType *int) ([]*Channel, int64, error)
	ListActive() ([]*Channel, error)
	ListActiveForUser(userID string) ([]*Channel, error)
	GetActiveByModel(model string) ([]*Channel, error)
	GetActiveByModelForUser(model string, userID string) ([]*Channel, error)
	UpdateStats(id string, success bool, quota int64) error
	UpdateStatsBatch(id string, requestCount int64, successCount int64, quota int64) error
}
